pytorch_lightning/utilities/device_parser.py

```diff
@@ -210,25 +212,25 @@ def _check_data_type(device_ids: Any) -> None:
         raise MisconfigurationException("Device ID's (GPU/TPU) must be int, string or sequence of ints or None.")
 
 
-def _tpu_cores_valid(tpu_cores):
+def _tpu_cores_valid(tpu_cores: Any) -> bool:
     # allow 1 or 8 cores
     if tpu_cores in (1, 8, None):
         return True
 
     # allow picking 1 of 8 indexes
     if isinstance(tpu_cores, (list, tuple, set)):
         has_1_tpu_idx = len(tpu_cores) == 1
-        is_valid_tpu_idx = tpu_cores[0] in range(1, 9)
+        is_valid_tpu_idx = list(tpu_cores)[0] in range(1, 9)
 
         is_valid_tpu_core_choice = has_1_tpu_idx and is_valid_tpu_idx
         return is_valid_tpu_core_choice
 
     return False
 
 
-def _parse_tpu_cores_str(tpu_cores):
+def _parse_tpu_cores_str(tpu_cores: str) -> Union[int, List[int]]:
     if tpu_cores in ('1', '8'):
-        tpu_cores = int(tpu_cores)
+        tpu_cores_int: Union[int, List[int]] = int(tpu_cores)
     else:
-        tpu_cores = [int(x.strip()) for x in tpu_cores.split(',') if len(x) > 0]
-    return tpu_cores
+        tpu_cores_int = [int(x.strip()) for x in tpu_cores.split(',') if len(x) > 0]
+    return tpu_cores_int
```

Comment:
Could just early exist to avoid redefinition

```suggestion
        return = int(tpu_cores)
    return [int(x.strip()) for x in tpu_cores.split(',') if len(x) > 0]
```

Reply:
Nice. This is definitely better! :)